src/main/java/io/cdap/plugin/gcp/gcs/sink/DelegatingGCSOutputCommitter.java

```diff
@@ -95,9 +131,37 @@ public void commitTask(TaskAttemptContext taskAttemptContext) throws IOException
 
   @Override
   public void commitJob(JobContext jobContext) throws IOException {
-    for (OutputCommitter committer : committerMap.values()) {
+    Path outputPath = new Path(jobContext.getConfiguration().get(DelegatingGCSOutputFormat.OUTPUT_PATH_BASE_DIR));
+    FileSystem fs = outputPath.getFileSystem(jobContext.getConfiguration());
+    Path tempPath = new Path(outputPath, FileOutputCommitter.PENDING_DIR_NAME);
+    Set<String> outputPaths = new HashSet<>();
+
+    for (FileStatus status : fs.listStatus(tempPath)) {
+      if (status.getPath().getName().endsWith("_partitions.txt")) {
+        FSDataInputStream dis =  fs.open(status.getPath());
+        DataInputStream in = new DataInputStream(new BufferedInputStream(dis));
+        BufferedReader br = new BufferedReader(new java.io.InputStreamReader(in));
+        String line;
+        while ((line = br.readLine()) != null) {
```

Comment:
There should never be more than 1 line

Reply:
It can have multiple lines if that split has records for multiple table names.

Reply:
one task can be responsible for processing multiple partitions, hence there can be multiple files.

Reply:
Got you. Well, in this case we need to ensure file is removed on the beginning of attempt (in case of any attempt retries)